libs/utils/src/lib/location-ext.service.ts

```diff
@@ -0,0 +1,44 @@
+import { Injectable, OnDestroy } from '@angular/core';
+import { Router, NavigationEnd, NavigationExtras } from '@angular/router';
+import { filter, skip, take } from 'rxjs/operators';
+import { Location } from '@angular/common';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class LocationExtService implements OnDestroy {
```

Comment:
I think "RouterNavigationService" is better. The service doesn't extend the location service. The only public method has the signature of `router.navigate`.